_posts/2023-11-15-Codesgiving-Part-1.md

```diff
@@ -0,0 +1,289 @@
+  var spec = {
+    "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
+    "description": "Suggested Open-source Time Allocations",
+    "width": 300,
+    "height": 300,
+    "data": {
+      "values": [
+        {"task": "Planning", "percentage": 33},
+        {"task": "Coding", "percentage": 16},
+        {"task": "Component and System Testing", "percentage": 25},
+        {"task": "Code Review, Revisions, and Post-Actions", "percentage": 25}
+      ]
+    },
+    "mark": {"type": "arc", "tooltip": true},
+    "encoding": {
+      "theta": {"field": "percentage", "type": "quantitative", "title": "Percentage"},
+      "color": {"field": "task", "type": "nominal", "title": "Task"}
+    }
+  };
+
+  const embed_opt = {"mode": "vega-lite"};
+  const el = document.getElementById('vis');
+  const view = vegaEmbed("#vis", spec, embed_opt);
+</script>
+_Suggested ratio of time spent by type of work for an open-source contribution._
+{:.center}
+
+The following modified rule of thumb from [_The Mythical Man Month_](https://www.oreilly.com/library/view/mythical-man-month-the/0201835959/) can assist with how you structure your time for an open-source contribution.
+Notice the emphasis on planning and testing and keep these in mind as you progress (the actual programming time can be small if adequate time has been spent on planning).
+Notably, the original time fractions are modified here with the final quarter of the time spent suggested as code review, revisions, and post-actions.
+Planning for the time expense of the added code review and related elements assists with keeping a learning mindset throughout the process (instead of feeling like the review is a "tack-on" or "optional / supplementary").
+A good motto to keep in mind throughout this process is [_Festina lente_](https://en.wikipedia.org/wiki/Festina_lente), or __"Make haste, slowly."__ (take care to move thoughtfully and as slow as necessary to do thing correctly the first time).
+
+1. 1/3 planning (~33%)
+2. 1/6 coding (~16%)
+3. 1/4 component and system testing (25%)
+4. 1/4 code review, revisions, and post-actions (25%)
+
+## Planning an Open-source Contribution
+
+### Has the Need Already Been Reported?
+
+Be sure to check whether the bug or feature has already been reported somewhere!
+In a way, this is a practice of ["Don't repeat yourself" (DRY)](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) where we attempt to avoid repeating the same block of code (in this case, the "code" can be understood as natural language).
+For example, you can look on GitHub Issues with a search query matching the rough idea of what you're thinking about.
+If it has been reported already, take a look to see if someone has made a code contribution related to the work already.
+If there aren't yet any code contributions and it doesn't look like anyone is working on one, consider volunteering to take a further look into the solution and be sure to acknowledge the existing discussions.
```

Comment:
Another very minor point, but I find that it helps to acknowledge the design difficulties and constraints of the author/maintainer, e.g. "i can see how creating a flexible and simple API would be hard for this, but maybe you could do something like this".

Reply:
Love this, showing sympathy or empathy can go a long way. Thanks - I'll find a way to add this in.